### ISSUE TITLE
Enable CORS for all endpoints

### ISSUE THREAD

**Issue by breandr** [joostfarla/serverless-cors-plugin#21]
It's my understanding that in order to add CORS to all endpoints you need to add `cors` to each function's `custom` property. Since the general use-case is that you want to either enable CORS on your entire API, or not at all, it would be nice if we could set this at a higher level that would be applied to all endpoints. We could allow setting the local `cors` property to override the global setting for the rare case you need it to be different/disabled.

If this is currently possible, I apolgise, and would greatly appreciate being pointed at documentation for this. :)

**Comment by erikerikson** [comment]
In case someone finds this issue too...
@breandr It is.  Just create a s-templates.json file at the root of your project and give it something like:

```
{
  "corsTemplate": {
    "allowOrigin": "*",
    "allowHeaders": ["Content-Type", "X-Amz-Date", "Authorization", "X-Api-Key"]
  }
}
```

Then in your s-function.json:
```
{
  ...
  "custom": {
    "cors": "$${corsTemplate}"
  }
  ...
}
```

**Comment by erikerikson** [comment]
If you want to override it in a single or subset of functions, you can either create a s-templates.json as a peer to your s-function.json that redefines `corsTemplate` or you can just not use the template at all.

**Comment by breandr** [comment]
Thanks, that's good to know. But what I was really hoping for was a "define once" solution. The template helps with reducing duplication, but ideally there would be no duplication and the functions would just inherit from the `custom` property in `s-project.json`. I'm new to serverless and not familiar with the patterns, so maybe this is a request for serverless core.

**Comment by joostfarla** [comment]
@breandr previously (SLS < 0.5), you could also set the policy on the component level. Since the component concept has been dropped, it was required to set the policy on each function (with or without using templates). 

I've just added support for setting the configuration on the project level. It's available in `v0.4.1`!

**Comment by breandr** [comment]
So good! Thanks mate.
